spring-rabbit/src/main/java/org/springframework/amqp/rabbit/AsyncRabbitTemplate.java

```diff
@@ -328,8 +330,9 @@ public void setReceiveTimeout(long receiveTimeout) {
 	 * @param taskScheduler the task scheduler
 	 * @see #setReceiveTimeout(long)
 	 */
-	public void setTaskScheduler(TaskScheduler taskScheduler) {
-		this.taskScheduler = taskScheduler; // NOSONAR synchronization
+	public synchronized void setTaskScheduler(TaskScheduler taskScheduler) {
+		this.internalTaskScheduler = false;
+		this.taskScheduler = taskScheduler;
```

Comment:
`Assert.notNull()` is missed.
Otherwise `this.internalTaskScheduler == false` and `this.taskScheduler == null` - and, therefore, NPE in the target code.
Test-case to be sure what we do ?